src/main/java/entities/Tag.java

```diff
@@ -0,0 +1,42 @@
+package entities;
+
+import java.util.ArrayList;
+
+public class Tag {
+    // The name of the tag
+    final String name;
+
+    // Array of posts that are under this tag
+    final ArrayList<Post> taggedPosts;
+
+    /**
+     * Initializes an instance of Tag.
+     *
+     * @param name The name of this tag.
+     */
+    public Tag(String name) {
+        this.name = name;
+        this.taggedPosts = new ArrayList<>();
+    }
+
+    /**
+     * Add a post that has this tag.
+     *
+     * @param post The post that has this name as a tag.
+     */
+    public void addTag(Post post) {
+        taggedPosts.add(post);
+    }
+
+    /**
+     * Returns whether the post that has this tag was successfully removed.
+     */
+    public boolean removeTag(Post post) { return taggedPosts.remove(post); }
```

Comment:
Consider renaming the method to something like removePost since we are removing a post, not a tag. Also, I think it is necessary to iterate through the taggedPosts, use the .equals method for Posts (overridden in Post entity; it compares posts based on their ID), and then remove the Post if its ID is equal to the ID of the Post that is to be deleted. This way there will be no issue with whether two posts that have the same id actually refer to the same object in memory.